Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts

```diff
@@ -268,20 +305,177 @@ namespace gdjs {
       );
     }
 
-    updateXPosition(): void {
-      if (!this._object._absoluteCoordinates) {
-        this._graphics.position.x = this._object.x;
-      } else {
+    invalidateBounds() {
+      this._object.invalidateBounds();
+      this._positionXIsUpToDate = false;
+      this._positionYIsUpToDate = false;
+    }
+
+    updatePreRender(): void {
+      this.updatePositionIfNeeded();
+    }
+
+    updatePositionX(): void {
+      if (this._object._useAbsoluteCoordinates) {
+        this._graphics.pivot.x = 0;
         this._graphics.position.x = 0;
+      } else {
+        // Make the drawing rotate around the rotation center.
+        this._graphics.pivot.x = this._object.getRotationCenterX();
+        // Multiply by the scale to have the scale anchor
+        // at the object position instead of the center.
+        this._graphics.position.x =
+          this._object.x +
+          this._graphics.pivot.x * Math.abs(this._graphics.scale.x);
       }
+      this._transformationIsUpToDate = false;
     }
 
-    updateYPosition(): void {
-      if (!this._object._absoluteCoordinates) {
-        this._graphics.position.y = this._object.y;
-      } else {
+    updatePositionY(): void {
+      if (this._object._useAbsoluteCoordinates) {
+        this._graphics.pivot.y = 0;
         this._graphics.position.y = 0;
+      } else {
+        this._graphics.pivot.y = this._object.getRotationCenterY();
+        this._graphics.position.y =
+          this._object.y +
+          this._graphics.pivot.y * Math.abs(this._graphics.scale.y);
+      }
+      this._transformationIsUpToDate = false;
+    }
+
+    updatePositionIfNeeded() {
+      if (!this._positionXIsUpToDate) {
+        this.updatePositionX();
+        this._positionXIsUpToDate = true;
+      }
+      if (!this._positionYIsUpToDate) {
+        this.updatePositionY();
+        this._positionYIsUpToDate = true;
+      }
+    }
+
+    updateTransformationIfNeeded() {
+      if (!this._transformationIsUpToDate) {
+        this.updatePositionIfNeeded();
+        this._graphics.updateTransform();
+      }
```

Comment:
@D8H We're missing a `this._transformationIsUpToDate = true` here no?

Reply:
I've added it and fixed an issue where setting center was not updating the transform :)